### ISSUE TITLE
 [Editor] Add loading spinner for image drag-and-drop upload

### ISSUE THREAD

**Issue by lisasy** [forem/forem#12443]
**Describe the bug**

Today, when someone in the editor (`/new`) drags-and-drops an image to upload it, they have no indication that it is currently uploading because there is no spinning indicator. This UX bug sets incorrect expectations on how our product works. 

**To Reproduce**

1. Go to `/new`
2. Drag an image file into body area to upload an image
3. Wait for image to upload 

This is how it incorrectly works here:

![current image upload](https://media.giphy.com/media/BdWSi27U4fgtGlEnz5/giphy.gif)

**Expected behavior**

![2021-01-26_Editor-Upload-Image-Fix](https://user-images.githubusercontent.com/816402/105910176-7200d380-5fdd-11eb-83cc-ecb7ed94d1f9.jpg)

The easiest solution here would be to render the same "Loading..." indicator when people upload an image from the "Upload image" button.

**Desktop (please complete the following information):**

- OS, version: Mac OS 11
- Browser, version: Firefox

**Comment by nickytonline** [comment]
Agree @lisasy. That is what I currently do for the cover image. As soon as you drop it, the spinner kicks in until the upload is finished. Not sure why I didn't do this for images in the content body of a post.

**Comment by vaidehijoshi** [comment]
@nickytonline quick question -- I saw you are assigned to this but also noticed it never made it into our current cycle work. Are you still working on/investigating it? I just added it to our issues to review in the meantime, but wanted to drop a note here to make sure I'm not missing anything as well :)

**Comment by nickytonline** [comment]
@vaidehijoshi, I had assigned it to me, because one of the folks pairing on the DEV stream was potentially going to work on it but they chose something else. I have someone coming on the stream next week that would like to pair on this.

**Comment by vaidehijoshi** [comment]
Sounds good -- I can take it out of the funnel in that case since you have a plan for it 👍🏽

**Comment by sethburtonhall** [comment]
Hey @nickytonline, reviewing this issue and looking forward to the stream this week.

**Comment by nickytonline** [comment]
Awesome. I've assigned it to you.

**Comment by Link2Twenty** [comment]
I've just given this a little think. I think probably the easiest solution (and it might actually be useful down the line too) is to move `useDragAndDrop` and `usePasteImage` from [EditorBody](https://github.com/forem/forem/blob/master/app/javascript/article-form/components/EditorBody.jsx) and into [ImageUploader](https://github.com/forem/forem/blob/master/app/javascript/article-form/components/ImageUploader.jsx). You'd have to pass in a reference to the element you wanted to listen on as a prop but that's it I think.

**Comment by mikeydorje** [comment]
Closing this as stale/outdated: the original no-feedback problem is now partially resolved since drag-and-drop uploads show an “Uploading image” text indicator (and button uploads show a spinner). The implementation has changed significantly since 2021, and there hasn’t been recent activity here. If we want to revisit and unify upload feedback UX, we can open a fresh issue with current context.
